### PR TITLE
CDAP-13622 add instance level profiles endpoints

### PR DESCRIPTION
**Pull request by albertshau**
Adding instance level profile endpoints to handle system profiles.
The namespaced endpoints for the system namespace will be modified
to disallow operations once the UI has switched over to these new
APIs.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java [hunk 1]
```diff
@@ -72,11 +72,63 @@ public ProfileHttpHandler(ProfileService profileService, ProvisioningService pro
...
+    profileService.deleteProfile(NamespaceId.SYSTEM.profile(profileName));
+    responder.sendStatus(HttpResponseStatus.OK);
+  }
+
   /**
    * List the profiles in the given namespace. By default the results will not contain profiles in system scope.
    */
   @GET
-  @Path("/profiles")
+  @Path("/namespaces/{namespace-id}/profiles")
```
Comment (yaojiefeng) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java:131]: Do we still allow accessing the system level profile using these namespaced REST endpoint?

**Review by yaojiefeng — APPROVED** [review]
Just one question about the validation, other things lgtm

**Comment by rohitsinha54** [comment]
It will be good to add specific entry in RouterPathLookup for profile endpoints to go to app-fabric. Currently it is going as it doesn't match any other case and also add unit test for profile endpoints in RouterPathLookupTest

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java (hunk 1 above)
Comment (albertshau) [reply cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java:131]: for the time being yes. I wanted to wait for UI to move off of those before adding a check to disallow.

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT6519

**Comment by albertshau** [comment]
I merged this without refreshing, didn't see Rohit's comment. Will do that in a follow up.

**Pull request merged**